addons/im_livechat/models/chatbot_mail_message.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+
+
+class ChatbotMailMessage(models.Model):
+    """ Chatbot Mail Message
+        We create a new model to store the related step to a mail.message and the user's answer.
+        We do this in a new model to avoid bloating the 'mail.message' model.
+    """
+
+    _name = 'im_livechat.chatbot.mail.message'
+    _description = 'Chatbot Mail Message'
+
+    mail_message_id = fields.Many2one('mail.message', string='Related Mail Message')
+    mail_channel_id = fields.Many2one('mail.channel', string='Related Mail Channel', required=True)
+    chatbot_step_id = fields.Many2one('im_livechat.chatbot.script_step', string='Chatbot Step', required=True, ondelete='cascade')
+    chatbot_question_answer_id = fields.Many2one('im_livechat.chatbot.script_question_answer', string="User's answer")
+    chatbot_user_raw_answer = fields.Text(string="User's raw answer")
```

Comment:
``user_raw_answer`` ? Note that this is not the raw answer but the "transformed" answer. Shoudln't we instead store the real mail.message body ? And maybe a computed with the found answer in case of selection (if necessary) ?

Reply:
Marking as resolved, has been reworked by Patrick to actually store the raw answer.